### DISCUSSION TITLE
Set the border color with JIT.

### DISCUSSION THREAD

**Discussion by es-ki** [tailwindlabs/tailwindcss#5696] [Help] [answered]
Is there a way to set the border color in JIT mode?  
I want to set the color specified by the user as the color of the border.  
The following code did not work.
```
const borderColor: string = getUserBorderColor();

<div className={`border-4 border-solid border-[${borderColor}]`}>
```

So far I've only faced this challenge in the border color case, but I have a feeling there are other cases like this.

**Comment by simonswiss** [comment, marked as answer]
Hey @Sakaki-Ya 👋

Just like in the traditional (AOT) mode in Tailwind CSS, with JIT you need to write entire class strings, so they are not "ignored" when the CSS is generated:

https://tailwindcss.com/docs/optimizing-for-production#writing-purgeable-html

JIT will not evaluate your code and determine which `border-*` color class to generate – it needs to be a real Tailwind CSS class (or arbitrary value).

It `userBorderColor` is a finite list of available colors, you can create a lookup table for it:

```js
const userBorderColors = {
  'blue': 'border-blue-500',
  'red': 'border-red-500',
  // ...
}
```

Then, you can do:

```html
<div className={`border-4 border-solid ${userBorderColors[borderColor]}`}>
```

If the color can be _anything_ (color picker), you're better off setting the border color via the style attribute:

```html
<div className="border-4 border-solid" style={{ borderColor: $borderColor }}
```

Hope it helps!
